libraries/primitive_after.rb

```diff
@@ -0,0 +1,57 @@
+require_relative 'primitive'
+
+module Choregraphie
+  class After < Primitive
+    def initialize(&block)
+      @block = block
+    end
+
+    def register(choregraphie)
+      @choregraphie_name = choregraphie.name
+
+      choregraphie.before do
+        set_in_progress
+      end
+
+      choregraphie.cleanup do
+        if in_progress?
+          @block.call
+          set_not_in_progress
+        elsif !installed? # we are after a reinstallation, we can't know if we were in a choregraphie or not, run the block just in case
+          @block.call
+        end
+        set_installed
+      end
+    end
+
+    def in_progress?
+      File.exist?(inprogress_marker)
+    end
+
+    def set_in_progress
+      FileUtils.touch(inprogress_marker) unless in_progress?
```

Comment:
Do you really need the guard? You want to put the marker always, and the implementation works even if the file is already present (as a bonus, you'll simplify the associated tests)

Reply:
good remark. The only reason I've put this is to have an implicit record (update date) of the beginning of the choregraphie. But you're right